module/signature/signer_indices.go

```diff
@@ -224,13 +225,19 @@ func DecodeSignerIndicesToIdentifiers(
 	// the signerIndices creator and validator see the same list.
 	signerIndices, err := CompareAndExtract(canonicalIdentifiers, prefixed)
 	if err != nil {
-		return nil, fmt.Errorf("could not extract signer indices from prefixed data: %w", err)
+		if errors.Is(err, ErrInvalidChecksum) {
+			return nil, NewInvalidSignerIndicesErrorf("signer indices' checkum is invalid: %w", err)
+		}
+		return nil, fmt.Errorf("unexpected exception while checking signer indices: %w", err)
 	}
 
 	numberCanonicalNodes := len(canonicalIdentifiers)
 	err = validPadding(signerIndices, numberCanonicalNodes)
 	if err != nil {
-		return nil, fmt.Errorf("signerIndices are invalid: %w", err)
+		if errors.Is(err, ErrIncompatibleBitVectorLength) || errors.Is(err, ErrIllegallyPaddedBitVector) {
+			return nil, NewInvalidSigTypesErrorf("invalid padding of signerIndices: %w", err)
```

Comment:
Why is this `NewInvalidSigTypesErrorf` rather than `NewInvalidSignerIndicesErrorf`?

Reply:
This is for the function `DecodeSigTypeToStakingAndBeaconSigners`. This function does _not_ decode the signer indices. To the best of my understanding, we call this function _after_ the signer indices were decoded to identities (or identifiers). In a second step, we decode the signature types (random beacon sig or staking sig) for each signer. Hence, I included a different sentinel so we can differentiate the root cause of errors.  But it would also be fine to consider 'InvalidSignerIndicesError' as a high-level error also covering failure cases for decoding the `sigType`. 🤷 either is fine by me.

Reply:
Yeah, I think using `InvalidSignerIndicesError` is necessary, because the caller of `DecodeSignerIndicesToIdentities` always expect `InvalidSignerIndicesError`.